conf/gulp-tasks/template/templatetaskfactory.js

```diff
@@ -14,67 +14,87 @@ class TemplateTaskFactory {
   constructor (translationResolver, locale = 'en') {
     this.locale = locale;
     this.translationResolver = translationResolver;
-    if (locale === 'en') {
-      this.filenamePrecompiled = 'answerstemplates.precompiled.min.js';
-      this.filenameUMD = 'answerstemplates.compiled.min.js';
-      this.filenameIIFE = 'answerstemplates-iife.compiled.min.js';
-    } else {
-      this.filenamePrecompiled = `${locale}-answerstemplates.precompiled.min.js`;
-      this.filenameUMD = `${locale}-answerstemplates.compiled.min.js`;
-      this.filenameIIFE = `${locale}-answerstemplates-iife.compiled.min.js`;
-    }
-    this.precompileTemplates = this.precompileTemplates.bind(this);
-    Object.defineProperty(this.precompileTemplates, 'name', { value: `${locale}-precompileTemplates` });
 
-    this.bundleTemplatesUMD = this.bundleTemplatesUMD.bind(this);
-    Object.defineProperty(this.bundleTemplatesUMD, 'name', { value: `${locale}-bundleTemplatesUMD` });
+    const { PRECOMPILED, UMD, IIFE } = this._getFileNames();
+    this.filenamePrecompiled = PRECOMPILED;
+    this.filenameUMD = UMD;
+    this.filenameIIFE = IIFE;
 
-    this.bundleTemplatesIIFE = this.bundleTemplatesIIFE.bind(this);
-    Object.defineProperty(this.bundleTemplatesIIFE, 'name', { value: `${locale}-bundleTemplatesIIFE` });
+    this._bindTasks();
+    this._customizeTaskNames(locale);
+  }
 
+  _bindTasks () {
+    this.precompileTemplates = this.precompileTemplates.bind(this);
+    this.bundleTemplatesUMD = this.bundleTemplatesUMD.bind(this);
+    this.bundleTemplatesIIFE = this.bundleTemplatesIIFE.bind(this);
     this.minifyTemplatesIIFE = this.minifyTemplatesIIFE.bind(this);
-    Object.defineProperty(this.minifyTemplatesIIFE, 'name', { value: `${locale}-minifyTemplatesIIFE` });
-
     this.minifyTemplatesUMD = this.minifyTemplatesUMD.bind(this);
-    Object.defineProperty(this.minifyTemplatesUMD, 'name', { value: `${locale}-minifyTemplatesUMD` });
-
     this.cleanFiles = this.cleanFiles.bind(this);
+  }
+
+  /**
+   * Customize the gulp task names, e.g. "bound precompileTemplates" becomes "en-precompileTemplates".
+   * @param {string} locale
+   */
+  _customizeTaskNames (locale) {
+    Object.defineProperty(this.precompileTemplates, 'name', { value: `${locale}-precompileTemplates` });
```

Comment:
LLs here for all the `defineProperty` calls. I think our LL is about 80ish chars (although this is not codified anywhere)

Reply:
oh I thought it was 100 chars (this is also over 100 though)